modules/apps/transfer/keeper/msg_server_test.go

```diff
@@ -131,14 +139,26 @@ func (suite *KeeperTestSuite) TestMsgTransfer() {
 				suite.Require().NoError(err)
 				suite.Require().NotNil(res)
 				suite.Require().NotEqual(res.Sequence, uint64(0))
-
-				events := ctx.EventManager().Events()
-				suite.assertTransferEvents(events, coin, "memo")
 			} else {
 				suite.Require().Error(err)
 				suite.Require().Nil(res)
+			}
 
-				events := ctx.EventManager().Events()
+			// Verify events
+			events := ctx.EventManager().Events()
+			expEvents := map[string]map[string]string{
+				"ibc_transfer": {
+					"sender":   suite.chainA.SenderAccount.GetAddress().String(),
+					"receiver": suite.chainB.SenderAccount.GetAddress().String(),
+					"amount":   coin.Amount.String(),
+					"denom":    coin.Denom,
+					"memo":     "memo",
+				},
+			}
+
+			if hasEvents {
+				ibctesting.AssertEvents(&suite.Suite, expEvents, events)
```

Comment:
I am wondering if it would be possible to move this assertion to the happy path case where `tc.expPass` would be true and then we wouldn't need the extra boolean `hasEvents`... Because are they both (and `tc.expPass` and `hasEvents`) true when we would like to assert events or would there be any situation where `tc.expPass` would be false but `hasEvents` is true?

Reply:
Also some more places where this was the case: https://github.com/cosmos/ibc-go/pull/2830/files#diff-ce312850968149b915fa31bf5158acb380a7efe3931048052422b0c6bd92ab80R749-R789

Reply:
Cool makes sense, then will use `hasEvents` wherever necessary

Reply:
Done @colin-axner sorry for the delay :)